mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/RetentionConfiguration.java

```diff
@@ -0,0 +1,68 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.commons.configuration.Configuration;
+import org.apache.james.util.TimeConverter;
+
+import com.google.common.base.Preconditions;
+
+public class RetentionConfiguration {
+    public static final RetentionConfiguration DEFAULT = new RetentionConfiguration(Duration.ofDays(365));
+
+    public static RetentionConfiguration from(Configuration propertiesConfiguration) {
+        return Optional.ofNullable(propertiesConfiguration.getString("retentionPeriod"))
+            .map(string -> TimeConverter.getMilliSeconds(string, TimeConverter.Unit.DAYS))
```

Comment:
why the hell do you convert a duration to milliseconds and then construct a duration from it?
You can be sure that, depending on when you do the computation, the result won't be accurate.
Please be more careful about Duration handling

Reply:
I don't see at all what you mean.

I do String -> long (ms) -> Duration

BTW that way of dealing with duration is quitte classic in the code base. If we have better, we should plan a refactoring...

Reply:
The user write
```
retentionPeriod=2
```
In his config file.
What we do is to transform "2" (that means "2 days") in milliseconds then convert it back to a Duration.
The thing is: 2 days is already a Duration and we should not convert it to milliseconds first.
This TimeConverter is wrong.

Reply:
TimeConverter became a DurationParser, and returns Duration objects.